core/domain/rte_component_registry_test.py

```diff
@@ -67,23 +67,27 @@ def _validate_customization_arg_specs(self, customization_arg_specs):
             self.assertTrue(isinstance(ca_spec['description'], basestring))
             self.assertGreater(len(ca_spec['description']), 0)
 
-            # The default value might not pass validation checks (e.g. the
-            # Image component has a required field whose default value is
-            # empty). Thus, when checking the default value schema, we don't
-            # apply the custom validators.
             schema_utils_test.validate_schema(ca_spec['schema'])
-            self.assertEqual(
-                ca_spec['default_value'],
-                schema_utils.normalize_against_schema(
-                    ca_spec['default_value'], ca_spec['schema'],
-                    apply_custom_validators=False))
-
-            if ca_spec['schema']['type'] == 'custom':
-                obj_class = obj_services.Registry.get_object_class_by_type(
-                    ca_spec['schema']['obj_type'])
+            # Default value of 'url' customization arg in rich-text component
+            # Link cannot be normalized as it is empty value. Hence when
+            # normalizing default values, we skip url.
+            if ca_spec['name'] != 'url':
```

Comment:
In addition to @bansalnitish's advice, I also suggest using the object type instead for the conditional. That's probably more robust than using the name.

Reply:
Here default value is for url, not for sanitized_url object as seen [here](https://github.com/oppia/oppia/pull/6286/files#diff-b8678e2fd0c471c19e17ac843efef9f0R95). Another customization_arg_spec may have obj_type sanitized url but non-empty default value. So I feel using `name` maybe better.

Reply:
`name` still has the problem that it's not robust. People can use any name when developing.

I still suggest using the object type but you can do something like "check that the value either matches validation checks, or is None". Would that work?

Reply:
Done.